app/validators/email_validator.rb

```diff
@@ -0,0 +1,8 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if value.nil? || value.strip.empty?
+    unless /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i.match?(value)
```

Comment:
@exbinary went to see if any of our dependencies already have an email regex.

The Devise gem includes an email regex, but I think that one is way too permissive.

@exbinary pointed out that Ruby's standard library includes `URI::MailTo::EMAIL_REGEXP`, which may be good enough for us. What do you think?

Reply:
Oh, never mind — I see that `URI::MailTo::EMAIL_REGEXP` would permit the bad example described in the issue